Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift

```diff
@@ -823,6 +850,20 @@ public struct RenderNodeTranslator: SemanticVisitor {
                 return ContentRenderSection(kind: .content, content: discussionContent, heading: "Discussion")
             }
             
+            /// Returns whether the topic with the given identifier is available in one of the traits in `allowedTraits`.
+            func isAvailableInAllowedTrait(identifier: String) -> Bool {
+                guard let reference = contentCompiler.collectedTopicReferences[identifier] else {
```

Comment:
Not sure I understand this logic. Could you explain why you return true if you couldn't find a topic reference for this identifier?

Reply:
Is this because you are conservatively not filtering out content you don't have a definition for in this framework?

Reply:
Ah yes, I should clarify that in a comment, thanks for pointing that out. It's expected for `reference` to never be `nil` as it is a precondition for `contentCompiler` to have populated `collectionTopicReference` with the reference at this point. Otherwise, it is a bug in `contentCompiler`.

Reply:
In fact, we can actually add an [`assertionFailure`](https://developer.apple.com/documentation/swift/1539616-assertionfailure) so that we get a crash in debug builds, but not release.

Reply:
Actually, after taking a closer look, I'm wrong on this. It's possible for the content compiler to generate a reference identifier without registering it in `collectedTopicReferences`, for example when the reference is a non-doc URL (e.g., `https://` in which case it gets registered in `linkReferences` instead. I'll add a comment to clarify that.